### PR TITLE
Prevent TypeError when using built-in constants in `setOption`. Fixes #1538

### LINKED ISSUE #1538: Redis::setOption() expects parameter 2 to be string, int given
### Expected behaviour

The following code should work without any warnings
```
<?php declare(strict_types=1);

$redis = new \Redis();
$redis->setOption(\Redis::OPT_SERIALIZER, \Redis::SERIALIZER_PHP);
$redis->setOption(\Redis::OPT_SCAN, \Redis::SCAN_RETRY);
```
### Actual behaviour
`setOption` expects second parameter to be as string. But the built in constants are long. This causes error in strict mode.

    Redis::setOption() expects parameter 2 to be string, int given

### I'm seeing this behaviour on
- OS: Mac
- Redis:
- PHP: 7.3
- phpredis: 4.2.0

### I've checked
- [x] There is no similar issue from other users
- [x] Issue isn't fixed in `develop` branch

### PR DESCRIPTION
**Pull request by JoyceBabu**
In strict_type mode `Redis::setOption` requires the second argument
to be of type `string`. This throws a TypeError for some options,
even when using built-in constants

The type of `value` argument of setOption has been changed to
`mixed` to prevent the error.

### CODE REVIEW

File: redis_commands.c [hunk 1]
```diff
@@ -3911,12 +3911,15 @@ void redis_setoption_handler(INTERNAL_FUNCTION_PARAMETERS,
...
+            val_str = zval_get_string(val);
+            if (ZSTR_LEN(val_str) > 0) {
+                redis_sock->prefix = val_str;
+            } else {
+                zend_string_release(val_str);
             }
             RETURN_TRUE;
         case REDIS_OPT_READ_TIMEOUT:
-            redis_sock->read_timeout = atof(val_str);
+            redis_sock->read_timeout = (float)zval_get_double(val);
```
Comment (jrchamp) [inline redis_commands.c]: The test for reading the timeout option is failing:
https://github.com/phpredis/phpredis/blob/bfaa89a8ae9b6a82dd8f3156e150ae898ee201be/tests/RedisTest.php#L4844-L4850

Do we need to use `(float)` here or can we stick with the double like `atof()` would normally return?

Comment (JoyceBabu) [reply redis_commands.c]: Sorry, I thought `atof` returned a `float`. I have fixed it.

File: redis_commands.c [hunk 2]
```diff
@@ -3911,12 +3911,15 @@ void redis_setoption_handler(INTERNAL_FUNCTION_PARAMETERS,
                 zend_string_release(redis_sock->prefix);
                 redis_sock->prefix = NULL;
             }
-            if (val_str && val_len > 0) {
-                redis_sock->prefix = zend_string_init(val_str, val_len, 0);
+            val_str = zval_get_string(val);
```
Comment (JoyceBabu) [inline redis_commands.c]: Do we need this check here?

File: redis_commands.c [hunk 3]
```diff
@@ -3911,12 +3911,15 @@ void redis_setoption_handler(INTERNAL_FUNCTION_PARAMETERS,
...
+            if (ZSTR_LEN(val_str) > 0) {
```
Comment (JoyceBabu) [inline redis_commands.c]: @jrchamp Do we need this check here?

Comment (jrchamp) [reply redis_commands.c]: Sorry - I'm not as familiar with the zend code.

**Comment by michael-grunder** [comment]
Merged, thanks!

**Pull request merged**